internal/grpc/services/ocmshareprovider/ocmshareprovider.go

```diff
@@ -157,6 +159,54 @@ func getOCMEndpoint(originProvider *ocmprovider.ProviderInfo) (string, error) {
 	return "", errors.New("ocm endpoint not specified for mesh provider")
 }
 
+// federatedID creates a federated user id by
+// 1. stripping the protocol from the domain and
+// 2. base64 encoding the opaque id with the domain to get a unique identifier that cannot collide with other users
+func federatedID(id *userpb.UserId) *userpb.UserId {
```

Comment:
the internal/grpc/services/ocmshareprovider/ocmshareprovider.go has the exact same func